src/widgets/index/index.ts

```diff
@@ -341,11 +365,26 @@ const index = (props: IndexProps): Index => {
 
       localInstantSearchInstance = null;
       localParent = null;
+      helper!.removeAllListeners();
       helper = null;
 
       derivedHelper!.detach();
       derivedHelper = null;
     },
+
+    getWidgetState(uiState: UiState) {
+      const currentUiState = Object.keys(localUiState).length
+        ? { ...uiState, [this.getIndexId()]: localUiState }
+        : uiState;
+
+      return localWidgets
+        .filter(isIndexWidget)
```

Comment:
```suggestion
        .filter(widget => isIndexWidget(widget))
```

But if you are sure we won't change the `isIndexWidget` signature, this is fine I guess

Reply:
TypeScript will complain anyway if we change the signature.